sfepy/discrete/materials.py

```diff
@@ -110,42 +119,49 @@ def __init__(self, name, kind='time-dependent',
         """
         Struct.__init__(self, name=name, kind=kind, is_constant=False)
 
-        if (function is not None) and ((values is not None) or len(kwargs)):
-            msg = 'material can have function or values but not both! (%s)' \
-                  % self.name
+        if kwargs:
+            if values is None:
+                values = kwargs
+            else:
+                values = dict(values)
+                values.update(kwargs)
+
+        if (function is not None) and (values is not None):
+            msg = f'material can have function or values but not both! ({self.name})'
```

Comment:
Let us use here also the `f'material {self.name}: ...` form of the message. E.g.
```
msg = f'material {self.name}: specify function or values but not both!'
```